spec/unit/providers/vsphere_spec.rb

```diff
@@ -691,6 +690,43 @@
     end
   end
 
+  describe '#set_network_device' do
+    let(:datacenter_object) { mock_RbVmomi_VIM_Datacenter({
+      :datastores => ['datastore0'],
+      :networks => ['network0'],
+    })
+    }
+    let(:template_vm_network_device)  { mock_RbVmomi_VIM_VirtualVmxnet3() }
+
+    before(:each) do
+      allow(subject).to receive(:connect_to_vsphere).and_return(connection)
+      allow(connection.serviceInstance).to receive(:find_datacenter).and_return(datacenter_object)
+    end
+
+    context 'Given an invalid network name' do
+      network_name = "invalid_network"
+
+      it 'should raise an error' do
+        expect { subject.set_network_device("datacenter_name", "template_vm_network_device", network_name, connection)}.to raise_error(/Cannot find network/)
```

Comment:
```suggestion
        expect { subject.set_network_device("datacenter_name", template_vm_network_device, network_name, connection)}.to raise_error(/Cannot find network/)
```
unless you meant to pass a string for the second argument?

Reply:
It didn't matter since the code would hit the raise error line before having to evaluate that variable but I'll remove the quotes